packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -764,6 +764,40 @@ const fn = new lambda.Function(this, 'MyFunction', {
 See [the AWS documentation](https://docs.aws.amazon.com/lambda/latest/dg/lambda-x-ray.html)
 to learn more about AWS Lambda's X-Ray support.
 
+## Lambda with AWS Distro for OpenTelemetry layer
+
+To have automatic integration with XRay without having to add dependencies or change your code, you can use the
+[AWS Distro for OpenTelemetry Lambda (ADOT) layer](https://aws-otel.github.io/docs/getting-started/lambda).
+Consuming the latest ADOT layer can be done with the following snippet:
+
+```ts
+import {
+  AdotLambdaExecWrapper,
+  AdotLambdaLayerJavaScriptSdkVersion,
+} from 'aws-cdk-lib/aws-lambda';
+
+const fn = new lambda.Function(this, 'MyFunction', {
+  runtime: lambda.Runtime.NODEJS_18_X,
+  handler: 'index.handler',
+  code: lambda.Code.fromInline('exports.handler = function(event, ctx, cb) { return cb(null, "hi"); }'),
+  adotInstrumentationConfig: {
```

Comment:
We should avoid confusing users about two slightly different concepts here: *ADOT instrumentation* and *ADOT Lambda layer*. Enabling ADOT instrumentation means two independent activities:
1. Adding an ADOT Lambda layer to the Lambda function.
2. Configuring the Lambda function to use an exec wrapper that's compatible with ADOT.

So setting a value for the exec wrapper means configuring the instrumentation, not configuring the layer.  I'm afraid that the proposed usage suggests the opposite. 

What do you think about this?

```ts
const fn = new lambda.Function(this, 'MyFunction', {
  ...
  adotInstrumentationConfig: AdotInstrumentationConfig.forJavaScriptSdk(
    AdotLambdaLayerJavaScriptSdkVersion.V1_7_0,
    AdotLambdaExecWrapper.REGULAR_HANDLER, // optional, defaulted to REGULAR_HANDLER
  ),
  ...
});
```
One thing I don't like about this approach though, is that we will have to introduce 5 more helper functions under `AdotLambdaConfig`. But I'm happy to go with it if it's right for users.

Reply:
Ok sure lets go with `adotInstrumentation` then (drop the `Config`).

Reply:
Sure! I've added a new commit to address this.

Reply:
I'm a bit skeptical with that experience. When I tried that new change in the code, and write this line as a user:

```ts
layerVersion: lambda.AdotLayerVersion.fromJavaSdkLayerVersion(lambda.AdotLambdaLayerJavaSdkVersion.LATEST),

```

... two things stood out for me:

1. When I typed up to `layerVersion: lambda.`, the auto-complete feature suggested me a bunch of options. The SDK-specific options are showed up at the top and `AdotLayerVersion`, which is the only one I can use, is near to the bottom. Most of the times, users know what SDK they need, so they would tend to choose those SDK-specific option first and get stuck. 
2. I have to type the same Layer type (i.e., `JavaSdk` ) twice on that line. 

---

Given those concerns, if you think they are minor enough, I'm happy to update the code to align with that experience.

Reply:
But my editor will show me that `layerVersion` takes an `AdotLayerVersion` type so I can just type that in and when I get to the `.` it will show me all the options. The alternative that you propose is that it would show me that it takes a type of `AdotLambdaLayerVersion` and I have to just know which other classes implement that. Nothing  in the editor will help me out.


> I have to type the same Layer type (i.e., JavaSdk ) twice on that line.

Yeah I get that, but that's just the way the versions work right? If every language had the same versions then this would be easier. We use this model elsewhere, for example the [rds.DatabaseCluster.engine](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_rds.DatabaseCluster.html#engine)

Reply:
I've updated the code ([the diff on top of old commits](https://github.com/aws/aws-cdk/commit/d20be2ab05d74db6ba1c7e8cc006fd696eda1573#diff-23d1a5e447cf921c3bb1b55507c661f44dd53e5317c0a4d35ea1ea6ed5c9cc6d)) to make the UX aligned with what you proposed. The code is a bit different from your gist since I want to avoid exposing irrelevant props to users. Let me know if you need me to change anything else.